src/libmongoc/src/mongoc/mongoc-log.c

```diff
@@ -198,7 +199,17 @@ mongoc_log_default_handler (mongoc_log_level_t log_level, const char *log_domain
    pid = (int) _lwp_self ();
 #elif defined(__APPLE__)
    uint64_t tid;
+#if MAC_OS_X_VERSION_MAX_ALLOWED < 1060 || defined(__ppc__)
```

Comment:
macOS 10.7 appears to remove the restriction that `pthread_threadid_np` is defined only on x86. Compare the libc linked versions linked from https://opensource.apple.com/releases/.

[From 10.6.0](https://github.com/apple-oss-distributions/Libc/blob/Libc-583/pthreads/pthread.h#L475-L477):

```c
#if defined(__i386__) || defined(__x86_64__)
int		pthread_threadid_np(pthread_t,__uint64_t*);
#endif
```

[From 10.7.0](https://github.com/apple-oss-distributions/Libc/blob/Libc-763.11/pthreads/pthread.h#L481):
```c
int pthread_threadid_np(pthread_t,__uint64_t*) __OSX_AVAILABLE_STARTING(__MAC_10_6, __IPHONE_3_2);
```

To simplify, suggest always using `pthread_mach_thread_np` for if the min version of macOS <= 10.7:
```c
#if MAC_OS_X_VERSION_MIN_REQUIRED <= MAC_OS_X_VERSION_10_7
   // libc on macOS <= 10.7 does not support `pthread_threadid_np` for all architectures.
   mach_port_t tid = pthread_mach_thread_np (pthread_self ());
   pid = (int) tid;
#else
   uint64_t tid;
   pthread_threadid_np (0, &tid);
   pid = (int) tid;
#endif
```

Reply:
@kevinAlbs Thank you. Sure, this can be done, `pthread_mach_thread_np` will work. I will rebase the PR accordingly.

Reply:
@kevinAlbs I have updated the fix. On 10.7 we can use `pthread_threadid_np`, since there it is supported for all archs in fact, so < 10.7 works fine.
We need to use a numerical macro though, since `MAC_OS_X_VERSION_10_7` is not defined prior to 10.7, so the effect will otherwise be exactly the opposite of what is needed.

Reply:
P. S. Do you know which header declares `mach_port_t` and whether it gets pulled in?

Reply:
On 10.6, it appears to be defined in the [pthread.h header](https://github.com/apple-oss-distributions/Libc/blob/36e20149daacb6cfafb07c68636d9c33b5509331/pthreads/pthread.h#L114-L117).

Reply:
Great, then all good. Thank you for merging!